client/checkout/utils/upe.js

```diff
@@ -64,3 +70,140 @@ export const getPaymentIntentFromSession = (
 
 	return {};
 };
+
+/**
+ * Finds selected payment gateway and returns matching Stripe payment method for gateway.
+ *
+ * @return {string} Stripe payment method type
+ */
+export const getSelectedUPEGatewayPaymentMethod = () => {
+	const paymentMethodsConfig = getUPEConfig( 'paymentMethodsConfig' );
+	const gatewayCardId = getUPEConfig( 'gatewayId' );
+	let selectedGatewayId = null;
+
+	// Handle payment method selection on the Checkout page or Add Payment Method page where class names differ.
+
+	if ( null !== document.querySelector( 'li.wc_payment_method' ) ) {
+		selectedGatewayId = document
+			.querySelector( 'li.wc_payment_method input.input-radio:checked' )
+			.getAttribute( 'id' );
+	} else if (
+		null !== document.querySelector( 'li.woocommerce-PaymentMethod' )
+	) {
+		selectedGatewayId = document
+			.querySelector(
+				'li.woocommerce-PaymentMethod input.input-radio:checked'
+			)
+			.getAttribute( 'id' );
+	}
```

Comment:
This could be significantly simplified:
```suggestion
	const radio = document.querySelector( 'li.wc_payment_method input.input-radio:checked, li.woocommerce-PaymentMethod input.input-radio:checked' );
	if ( null !== radio ) {
		selectedGatewayId = radio.id;
	}
```

Also, is there any reason to use `getAttribute( 'id' )` instead of just `id`? It's a native HTML prop.

Reply:
Excellent suggestion, and there's no reason to use `getAttribute('id')` instead of `id`. I've copied this code from another place, but I fully agree that we better improve it as well. 749d7d9229b6076e3ce8b4bff8716a1da14b3270